source/module_elecstate/elecstate_energy.cpp

```diff
@@ -131,13 +131,16 @@ double ElecState::cal_delta_eband(const UnitCell& ucell) const
         if (get_xc_func_type() == 3 || get_xc_func_type() == 5)
         {
             v_ofk = this->pot->get_effective_vofk(0);
+            
         }
-
+        
         for (int ir = 0; ir < this->charge->rhopw->nrxx; ir++)
         {
             deband_aux -= this->charge->rho[0][ir] * (v_eff[ir] - v_fixed[ir]);
             if (get_xc_func_type() == 3 || get_xc_func_type() == 5)
```

Comment:
the if judgement seems strange, it should not be in the middle of for loop

Reply:
Add a bool flag "v_ofk_flag" for the judegment.